lib/menu_screen.dart

```diff
@@ -148,7 +150,11 @@ class _MenuScreenView extends StatelessWidget {
             ),
             ScreenButton(
               buttonText: "Run Analysis",
-              pressFunction: state.runAnalysis,
+              pressFunction: () {
+                Navigator.pushNamed(
+                  context,
+                  LoadingScreen.id);
+              },
```

Comment:
state.runAnalysis() is a function in the class above that encapsulates all the functionality to run an analysis. Instead of overwriting it, just add your code to push the loading screen to that function.

The runAnalysis() function also has a built-in check to see if any images are missing.